tiny_dnn/layers/layer.h

```diff
@@ -347,6 +353,67 @@ class layer : public node {
     return *this;
   }
 
+  /**
+   * @name Parameter Getters and Setters
+   * @{
+   */
+
+  /** @brief Add new parameter to this layer, to be called in constructor. */
+  void add_parameter(size_t out_channels,
+                     size_t in_channels,
+                     size_t height,
+                     size_t width,
+                     parameter_type type,
+                     bool trainable = true) {
+    parameters_.push_back(
+      new Parameter(out_channels, in_channels, height, width, type, trainable));
+  }
+
+  /**
+   * @brief Get pointers to parameters of this layer.
+   *
+   * This can be used during training phase when parameter update is
+   * required.
+   *
+   * @param trainable_only flag to return only the trainable parameters.
+   * @return std::vector of pointers to parameters
+   */
+  Parameters parameters(bool trainable_only = false) {
```

Comment:
@edgarriba I thought it might be useful when user is simply pulling out weights for visualizing weights or something. But I am okay with removing it...